fieldservice_sale/models/sale_order.py

```diff
@@ -229,3 +229,18 @@ def action_view_fsm_order(self):
         else:
             action = {"type": "ir.actions.act_window_close"}
         return action
+
+    def write(self, vals):
+        res = super().write(vals)
+        for order in self:
+            # FSM Orders are created when the sale order is confirmed.
+            # So we skip the commitment_date propagation if the sale order is in draft.
+            # That means that the FSM Orders are not created yet.
+            # We also skip if the sale order is cancelled as it makes no sense.
+            if (
+                order.state not in ("draft", "cancel")
+                and vals.get("commitment_date") is not None
+            ):
+                for fsm_order in order.fsm_order_ids:
+                    fsm_order.scheduled_date_start = vals["commitment_date"]
+        return res
```

Comment:
I think here we should align to what Odoo does in `sale_stock`: https://github.com/odoo/odoo/blob/6194ca0042f967308178699837f01d42f46dea52/addons/sale_stock/models/sale_order.py#L143-L148

Meaning:
- Sync even if the sale order is confirmed
- Fallback to order.expected_date if there's no commitment_date